### PR TITLE
Sets work order ID when adding schedule work tasks to follow up work orders

### PR DESCRIPTION
**Pull request by johnclary**
Resolves https://github.com/cityofaustin/atd-data-tech/issues/10207.

I had to take a slightly different approach from Mike's [recent implementation](https://github.com/cityofaustin/atd-data-tech/issues/9052#issuecomment-1142315949), because this form can be submitted multiple times. Also, b/c we're listening here for the `scene` instead of `view` render event, I didn't seem to run into any issues with chosen.js's own DOM manipulation.

You can test this in the `TEST | Austin Transportation Data Tracker | 27 OCT 2021` app on this [page](https://atd.knack.com/test--austin-transportation-data-tracker--27-oct-2021#work-orders/work-order-details/6318f9745b084d002443a64c/follow-up-work-order/6318f9745b084d002443a64c/scheduled-followup/6318f9745b084d002443a64c/)

1. Open that page
2. Confirm the follow-up work order is set in the **Work Order ID** field
3. Select a **Work Type** and submit the form
4. When the form reloads, confirm the follow-up work order is set in the **Work Order ID** field

Screen Shot 2022-09-08 at 1 08 12 PM

### CODE REVIEW

File: code/data-tracker/data-tracker.js [hunk 1]
```diff
@@ -1043,3 +1007,76 @@ $(document).on("knack-view-render.view_1146", function (event, view, data) {
...
+    $placeholderTextSpan.text(originalWorkOrderIdText);
+  });
+});
+
+//////////////////////////////////////////////////////////////////
+/// End Autopopulate work order ID in the follow-up order form ///
+//////////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////////////
```
Comment (johnclary) [inline code/data-tracker/data-tracker.js:1048]: this is where the new code begins - sorry about ☝️

**Review by dianamartin — APPROVED** [review]
I figure you're doing some clean up

**Review by dianamartin — APPROVED** [review]

**Review by dianamartin — COMMENTED** [review]
I test this function in "TEST | Austin Transportation Data Tracker | 27 OCT 2021" and it's worked, but when the form was submitted the work order ID was wiped. 

I had to manually refresh the the page for the work order ID was populated again.

**Comment by johnclary** [comment]
@dianamartin would you try again? looks like i didn't save the code correctly in the builder.

**Comment by dianamartin** [comment]
@johnclary I just re-tested
- [x] Added a new record
- [x] Watched the form refresh and populate the Work Order ID
- [x] Works like a charm!

**Comment by johnclary** [comment]
👍 shall i deploy in production?

**Comment by johnclary** [comment]
i will happily do it later. will need to get all the view ids updated in that case. just let me know

**Comment by dianamartin** [comment]
@johnclary Do we want to do this Friday afternoon?

**Comment by johnclary** [comment]
After testing this a bit more, it was not working consistently. The trick, which @mddilley already discovered, is that the connection `<select>` input _must_ have an option present that matches the value we're trying to set the input to. That should have been obvi, but for whatever reason the form was sometimes working correctly.

Anyway @dianamartin would you please test this again?

Also, as we discussed—the details view is currently showing the parent work order ID's _parent_ work order ID, which is the ID we're using when we auto-set the connection. Let me know if you want to switch that up and I'll update the code

Screen Shot 2022-09-12 at 12 19 28 PM

**Comment by dianamartin** [comment]
# 9/13
@johnclary 
Just tested the Scheduled Follow-up work order enhancement and it works. I updated the Details page to show the actual Work Order ID. I just can't get it to hyperlink, which is fine. 

Looks like this now:
image

**Review by mddilley — APPROVED** [review]
Test steps check out, and I like using the interval to make sure that value continually gets in the form. 👍  We are going to be experts in Chosen.js select hackery 😄🔥

**Comment by dianamartin** [comment]
@mddilley @johnclary what's the status of this, it was in the "New" pipeline for me as of 10/20/23.

**Comment by mddilley** [comment]
@johnclary I rediscovered this PR when working on https://github.com/cityofaustin/atd-data-tech/issues/15708. I think it is good to merge! The code is in the JS that I copied out of prod data tracker.

**Comment by johnclary** [comment]
This is covered by Mike's more recent work: https://github.com/cityofaustin/atd-knack/pull/153.

Closing.

**Pull request closed without merging**